src/googleclouddebugger/firebase_client.py

```diff
@@ -312,6 +321,19 @@ def _TransmissionThreadProc(self):
 
       self._new_updates.wait(delay)
 
+  def _MarkActiveTimerFunc(self):
+    """Entry point for the mark active timer."""
+
+    try:
+      self._MarkDebuggeeActive()
+    finally:
+      self._StartMarkActiveTimer()
```

Comment:
If an exception occurs it will propagate out. Is this safe? Or is this intended.

If you want to swallow the exception adding a return in the finally clause would do it I believe,https://docs.python.org/3/reference/compound_stmts.html#finally-clause

Either way a comment would be helpful

Reply:
Good question.  The function will only ever be run under its own thread, and the thread will terminate at the end of this function, so the main question is if/how the exceptions should be surfaced.

What are your thoughts here?

Reply:
I think in this case it would be reasonable to capture it and  emit a log warning with the error and that the update failed, but otherwise letting it reschedule for the next attempt and moving on. Also it seems _MarkDebuggeeActive catches exceptions also, so it seems the likelihood of one reaching here would be low, I'm not sure which other types might occur that _MarkDebuggeeActive wouldn't catch.

Reply:
Sounds reasonable; I'll do that.

And yes, this is extremely defensive coding.